File: etl-common/src/test/scala/org/bigdate/etl/common/scala/test/spark/executors/transform/DirtyTransformExecutor.scala
package org.bigdate.etl.common.scala.test.spark.executors.transform

import org.apache.spark.rdd.RDD
import org.apache.spark.sql.{DataFrame, SparkSession}
import org.bigdata.etl.common.annotations.ETLExecutor
import org.bigdata.etl.common.enums.CommonConstants
import org.bigdata.etl.common.executors.TransformExecutor
import org.bigdate.etl.common.scala.test.spark.configs.DirtyConfig
import org.slf4j.{Logger, LoggerFactory}

/**
 * Author: GL
 * Date: 2022-05-05
 */
@ETLExecutor("dirty")
class DirtyTransformExecutor extends TransformExecutor[SparkSession, RDD[String], DataFrame, DirtyConfig] {

  val log: Logger = LoggerFactory.getLogger(classOf[DirtyTransformExecutor])

  override def init(engine: SparkSession, config: DirtyConfig): Unit = {
    log.info("DirtyTransform init, config: {}", config)
  }

  override def process(engine: SparkSession, value: RDD[String], config: DirtyConfig): DataFrame = {
    import engine.implicits._
    val dirtyRdd = value.filter(v => v.split(CommonConstants.SPLIT_STRING).length == CommonConstants.FIRST)
    dirtyRdd.saveAsTextFile(config.dirtyPath) // 将脏数据存储至dirtyPath
    value.toDF()
  }

  override def close(engine: SparkSession, config: DirtyConfig): Unit = {
    log.info("DirtyTransformExecutor close, config: {}", config)
  }

  override def check(config: DirtyConfig): Boolean = {
    log.info("DirtyTransformExecutor check, config: {}", config)
    true
  }
}

File: etl-common/src/test/java/org/bigdata/etl/common/java/test/executors/transform/DirtyTransformExecutor.java
package org.bigdata.etl.common.java.test.executors.transform;

import lombok.extern.slf4j.Slf4j;
import org.apache.spark.SparkContext;
import org.apache.spark.rdd.RDD;
import org.bigdata.etl.common.annotations.ETLExecutor;
import org.bigdata.etl.common.java.test.configs.DirtyConfig;
import org.bigdata.etl.common.executors.TransformExecutor;


/**
 * Author: GL
 * Date: 2022-04-22
 */
@Slf4j
@ETLExecutor("dirty")
public class DirtyTransformExecutor implements TransformExecutor<SparkContext, RDD<String>, RDD<String>, DirtyConfig> {

    @Override
    public void init(SparkContext engine, DirtyConfig config) {
        log.info("DirtyTransform init, config: {}", config);
    }

    @Override
    public RDD<String> process(SparkContext engine, RDD<String> value, DirtyConfig config) {
        log.info("DirtyTransform process, config: {}, Transform collect: {}", config, value.collect());
        return value;
    }

    @Override
    public void close(SparkContext engine, DirtyConfig config) {
        log.info("DirtyTransformExecutor close, config: {}", config);
    }

    @Override
    public boolean check(DirtyConfig config) {
        log.info("DirtyTransformExecutor check, config: {}", config);
        return true;
    }
}
